agents/wiregrid_kikusui/kikusui_agent.py

```diff
@@ -0,0 +1,502 @@
+        self.log.warn(msg)
+        return False
+
+    def set_on(self, session, params = None):
+        with self.lock.acquire_timeout(0, job = 'set_on') as acquired:
+            if not acquired:
+                self.log.warn('Could not set ON because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            # wait for getting control
+            ret = self.__wait_control()
+            if ret :
+                self.cmd.user_input('on')
+                self.switching = False
+            else :
+                msg = 'Could not set ON because of failure in getting the control.'
+                self.log.warn(msg)
+                return False, msg
+                pass
+            pass
+
+        return True, 'Set Kikusui on'
+
+    def set_off(self, session, params = None):
+        with self.lock.acquire_timeout(0, job = 'set_off') as acquired:
+            if not acquired:
+                self.log.warn('Could not set OFF because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            # wait for getting control
+            ret = self.__wait_control()
+            if ret :
+                self.cmd.user_input('off')
+                self.switching = False
+            else :
+                msg = 'Could not set OFF because of failure in getting the control.'
+                self.log.warn(msg)
+                return False, msg
+                pass
+            pass
+
+        return True, 'Set Kikusui off'
+
+    def calibrate_itself(self, session, params = None):
+        if params == None:
+            params = {'storepath':self.action_path}
+            pass
+
+        with self.lock.acquire_timeout(0, job = 'set_off') as acquired:
+            if not acquired:
+                self.log.warn('Could not calibrate itself because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+        logfile = openlog(params['storepath'])
+
+        ret = self.__wait_control()
+
+        cycle = 1
+        for i in range(11):
+            tperiod = 0.10 + 0.02*i
+            for j in range(100):
+                if j%20 == 0:
+                    self.log.warn(f'this is {cycle}th time action')
+                    pass
+
+                writelog(logfile, 'ON', tperiod, self.get_position(self.position_path, self.open_trial, self.Deg))
+                self.rotate_alittle(ret, tperiod)
+                time.sleep(self.agent_interval+1.)
+                writelog(logfile, 'OFF', 0., self.get_position(self.position_path, self.open_trial, self.Deg))
+                writelog(logfile, 'ON', 0.70, self.get_position(self.position_path, self.open_trial, self.Deg))
+                self.rotate_alittle(ret, 0.70)
+                time.sleep(self.agent_interval+1.)
+                writelog(logfile, 'OFF', 0., self.get_position(self.position_path, self.open_trial, self.Deg))
```

Comment:
I want an isolated log file for debugging.
I add a debug option to turn ON the log file. It is OFF nominally.
By the way, in calibrate_wg() function, the writelog()s always occur because the calibrate_wg() is a command just for commissioning of the rotation motor and the log file is necessary to adjust the rotation control parameters (self.feedback_time).